src/core_codemods/with_threading_lock.py

```diff
@@ -39,8 +40,9 @@ def leave_With(self, original_node: cst.With, updated_node: cst.With):
         if len(original_node.items) == 1 and self.node_is_selected(
             original_node.items[0]
         ):
-            # TODO: how to avoid name conflicts here?
-            name = cst.Name(value="lock")
+            current_names = self.find_used_names(original_node)
+            value = "lock" if "lock" not in current_names else "lock_"
```

Comment:
hm just now realizing I should probably ALSO check that `lock_` isn't taken....

Reply:
This codemod could make several changes to the same scope (unlikely I guess, but possible). We could just look for `lock_*` until we get a miss (there are only finitely many names in a module, so it will always succeed).